### PR TITLE
docs: add docs for private endpoint connections

### PR DESCRIPTION
**Pull request by carloruiz**
Adds more detailed docs for the `private_endpoint_connection` resource.

**Commit checklist**
- [ ] Changelog
- [ ] Doc gen (`make generate`)
- [ ] Integration test(s)
- [ ] Acceptance test(s)
- [ ] Example(s)

### CODE REVIEW

File: docs/resources/private_endpoint_connection.md [hunk 1]
```diff
@@ -10,12 +10,80 @@ description: |-
 
 Private Endpoint Connection.
 
-## Example Usage
+Private endpoint connections allow customer applications to connect to a CockroachDB Cloud cluster without traversing the public internet. All application-database traffic remains within the cloud-provider network.
+
+**IMPORTANT:**You must have the following role to successfully handle the resource:
```
Comment (mikeCRL) [inline docs/resources/private_endpoint_connection.md:15]:
```suggestion
**IMPORTANT:** You must have the following role to successfully handle the resource:
```

Comment (carloruiz) [reply docs/resources/private_endpoint_connection.md]: done

File: docs/resources/private_endpoint_connection.md [hunk 2]
```diff
@@ -3,19 +3,82 @@
...
+
+## Example with Azure Private Link
+
+# Enable the Private Link service on the CockroachDB Cloud cluster.
+resource "cockroach_private_endpoint_services" "services" {
+  cluster_id = cockroach_cluster.my_cluster.id
+}
+
+# Create a private link endpoint and associate it with the Private Link Service. 
+resource "azurerm_private_endpoint" "my_endpoint" {
```
Comment (fantapop) [inline docs/resources/private_endpoint_connection.md:46]: is this right? `azurerm_`...?

Comment (carloruiz) [reply docs/resources/private_endpoint_connection.md:46]: Yiup https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/private_endpoint

**Review by caobill — APPROVED** [review]
lgtm

File: docs/resources/private_endpoint_connection.md (hunk 2 above)
Comment (fantapop) [reply docs/resources/private_endpoint_connection.md:46]: oh, got it.  It's not ours.  👍

File: examples/resources/cockroach_private_endpoint_connection/resource.tf [hunk 3]
```diff
@@ -1,4 +1,67 @@
...
+# Create the GCP Private Service Connect endpoint using the GCP API or the GCP
+# Console. You will need the service id to create the endpoint. You can get the
+# service information by running `terraform show` and noting
+# `cockroach_private_endpoint_connection.services[*].name`,
+# `cockroach_private_endpoint_connection.services[*].endpoint_service_id`
+
+# Establish a connection between the endpoint and the service.
+resource "cockroach_private_endpoint_connection" "connection" {
+  cluster_id  = cockroach_cluster.my_cluster.id
+  endpoint_id = "6133183410995353"
```
Comment (fantapop) [inline examples/resources/cockroach_private_endpoint_connection/resource.tf:66]: Could this just be:
`endpoint_id = azurerm_private_endpoint.my_endpoint.id`

**Review by fantapop — APPROVED** [review]
I left one suggestion but whatever your response there, this seems good to me.

File: examples/resources/cockroach_private_endpoint_connection/resource.tf (hunk 3 above)
Comment (carloruiz) [reply examples/resources/cockroach_private_endpoint_connection/resource.tf:66]: This particular declaration is for GCP, which doesn't have a TF resource for private endpoint connections.

**Pull request merged**